src/sim/models.py

```diff
@@ -52,9 +54,8 @@ def __init__(self, P0: float, rho: float):
         self.rho = rho
 
     def __call__(self, B, qE):
-        C = np.where(B > 0, B, 1.)  # set those entries in B that vanish to 1. arbitrarily
-        PB = np.where(B > 0, self.P0 * C ** self.rho, 0.)  # those entries where C is 1 are 0. in PB
-        return PB * qE * B
+        market_price = self.P0 * B ** self.rho
+        return market_price * qE * B
 
```

Comment:
with setting `Bs = np.maximum(1, Bs_step)` in the ode evolver, we don't need to worry about B==0 anywhere